src/Containers/Microsoft.NET.Build.Containers/Registry/DefaultManifestOperations.cs

```diff
@@ -41,7 +41,8 @@ public async Task PutAsync(string repositoryName, string reference, ManifestV2 m
 
         if (!putResponse.IsSuccessStatusCode)
         {
-            throw new ContainerHttpException(Resource.GetString(nameof(Strings.RegistryPushFailed)), putResponse.RequestMessage?.RequestUri?.ToString(), jsonString);
+            await putResponse.LogHttpResponseAsync(_logger, cancellationToken).ConfigureAwait(false);
+            throw new ContainerHttpException(Resource.FormatString(nameof(Strings.RegistryPushFailed), putResponse.StatusCode), putResponse.RequestMessage?.RequestUri?.ToString());
```

Comment:
instead of passing `jsonString` here we should be passing something like `await putResponse?.Content?.ReadAsStringAsync().ConfigureAwait(false)` so that we can get the payload returned by the server. These often have details that can help diagnose issues.